runtime/tchannel_client.go

```diff
@@ -171,7 +175,29 @@ func (c *TChannelClient) CallThruAltChannel(
 		metrics:       c.metrics[serviceMethod],
 	}
 
-	return c.call(ctx, call, reqHeaders, req, resp, true)
+	return c.call(ctx, call, reqHeaders, req, resp, true, "")
+}
+
+// CallToHostPort makes a RPC call using a dedicated hostport
+func (c *TChannelClient) CallToHostPort(
```

Comment:
This is unnecessary, the hostport should just be added as a peer when the client is created. No need to modify the `call` function either.

Reply:
client is created at init time with 1 channel for inbound request and 2 subchannel for outbound request. one of the subchannel is used for making request to prod, another one is altsubchannel for making request to staging. peer list only has 127.0.0.1:5437 at the begining